whitelist.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"net"
+)
+
+type IPWhitelist struct {
+	ip4Whitelist *[]*ipNetMap
+	ip6Whitelist *[]*ipNetMap
+}
+
+func NewIPWhitelist(nets []*net.IPNet) *IPWhitelist {
+	ip4Whitelist := make([]*ipNetMap, 0)
+	ip6Whitelist := make([]*ipNetMap, 0)
+
+	w := &IPWhitelist{
+		ip4Whitelist: &ip4Whitelist,
+		ip6Whitelist: &ip6Whitelist,
+	}
+
+	for _, net := range nets {
+		w.addIP(net.IP, net.Mask)
+	}
+
+	return w
+}
+
+func (w IPWhitelist) getNetMaps(ip net.IP) (netMaps *[]*ipNetMap) {
```

Comment:
```suggestion
func (w* IPWhitelist) getNetMaps(ip net.IP) (netMaps *[]*ipNetMap) {
```
Was using a non-pointer intentional for these "member" functions?

Reply:
Sort of, I didn't necessarily want to encourage modification of the ipNetMap pointers for the implementation of the `has` function.

But to be honest on a semi-related I got a little confused with how Golang is handling this pointers vs C, I had trouble getting the the address of the value of the slice in these methods, thus the crazy `*[]*ipNetMap` type.  I might have just been doing something silly though, but it felt weirdly different and Intuitive.  I feel I'm rarely using the reference operator in Golang and always the dereference.

Reply:
Fixed.  Figured it out, not sure why I was having issues.